pkg/kapp/diff/filter.go

```diff
@@ -0,0 +1,111 @@
+// Copyright 2020 VMware, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package diff
+
+import (
+	"encoding/json"
+
+	ctlres "github.com/k14s/kapp/pkg/kapp/resources"
+)
+
+type ChangeSetFilter struct {
+	Filter string
+}
+
+type OpsFilter []ChangeOp
+
+type ChangeSetFilterRoot struct {
+	And              []ChangeSetFilterRoot
+	Or               []ChangeSetFilterRoot
+	Not              *ChangeSetFilterRoot
+	Ops              OpsFilter
+	NewResource      *ctlres.ResourceFilter
+	ExistingResource *ctlres.ResourceFilter
+}
+
+func (s *ChangeSetFilter) DiffFilter() (ChangeSetFilterRoot, error) {
+	var filter ChangeSetFilterRoot
+	if len(s.Filter) > 0 {
+		changeSetFilter, err := NewChangeSetFilterFromString(s.Filter)
```

Comment:
Nit: Variable name can be changeSetFilterRoot as NewChangeSetFilterFromString returns changeSetFilterRoot

Reply:
Done